--- pbr-org-1.1.6-4.bash	2024-07-13 08:45:38.678916000 +0200
+++ pbr-egc-2-1.1.6-4.bash	2024-07-13 12:14:23.083037000 +0200
@@ -1227,15 +1227,18 @@
 # original idea by @egc112: https://github.com/egc112/OpenWRT-egc-add-on/tree/main/stop-dns-leak
 dns_policy_routing() {
 	local mark i nftInsertOption='add'
-	local param4_tcp param4_udp param6_tcp param6_udp
+	local param4 param4_tcp param4_udp param6 param6_tcp param6_udp
 	local negation value dest4 dest6
 	local dest_dns_interface dest_dns_ipv4 dest_dns_ipv6
 	local name="$1" src_addr="$2" dest_dns="$3" uid="$4"
 	local chain='dstnat_lan' iface='dns'
 
 	dest_dns_interface="$(str_first_value_interface "$dest_dns")"
-	dest_dns_ipv4="$(str_first_value_ipv4 "$dest_dns")"
-	dest_dns_ipv6="$(str_first_value_ipv6 "$dest_dns")"
+	if is_ipv4 "$dest_dns"; then
+		dest_dns_ipv4="$(str_first_value_ipv4 "$dest_dns")"
+	elif is_ipv6 "$dest_dns"; then
+		dest_dns_ipv6="$(str_first_value_ipv6 "$dest_dns")"
+	fi
 
 	if is_supported_interface "$dest_dns_interface"; then
 		local d
@@ -1289,8 +1292,11 @@
 			param4="${param4:+$param4 }${nftIPv4Flag} saddr ${negation:+$negation }{ $(inline_set_resolve_ipv4 "$value") }"
 			param6="${param6:+$param6 }${nftIPv6Flag} saddr ${negation:+$negation }{ $(inline_set_resolve_ipv6 "$value") }"
 		else
-			param4="${param4:+$param4 }${nftIPv4Flag} saddr ${negation:+$negation }{ $(inline_set "$value") }"
-			param6="${param6:+$param6 }${nftIPv6Flag} saddr ${negation:+$negation }{ $(inline_set "$value") }"
+			if is_ipv4 "$value"; then
+				param4="${param4:+$param4 }${nftIPv4Flag} saddr ${negation:+$negation }{ $(inline_set "$value") }"
+			elif is_ipv6 "$value"; then
+				param6="${param6:+$param6 }${nftIPv6Flag} saddr ${negation:+$negation }{ $(inline_set "$value") }"
+			fi
 		fi
 	fi
 
@@ -1299,8 +1305,12 @@
 	param6_tcp="$nftInsertOption rule inet ${nftTable} ${nftPrefix}_${chain} ${param6} tcp ${dest6} comment \"$name\""
 	param6_udp="$nftInsertOption rule inet ${nftTable} ${nftPrefix}_${chain} ${param6} udp ${dest6} comment \"$name\""
 	local ipv4_error='0' ipv6_error='0'
-	{ nft4 "$param4_tcp" && nft4 "$param4_udp"; } || ipv4_error='1'
-	{ nft6 "$param6_tcp" && nft6 "$param6_udp"; } || ipv6_error='1'
+	if [ -n "$param4" ] && [ -n "$dest_dns_ipv4" ]; then
+		{ nft4 "$param4_tcp" && nft4 "$param4_udp"; } || ipv4_error='1'
+	fi
+	if [ -n "$param6" ] && [ -n "$dest_dns_ipv6" ]; then
+		{ nft6 "$param6_tcp" && nft6 "$param6_udp"; } || ipv6_error='1'
+	fi
 	if [ -n "$ipv6_enabled" ] && [ "$ipv4_error" -eq '1' ] && [ "$ipv6_error" -eq '1' ]; then
 		processPolicyError='true'
 		state add 'errorSummary' 'errorPolicyProcessInsertionFailed' "$name"
