--- pbr-1.1.7-29.bash	2024-10-25 08:17:21.702349600 +0200
+++ pbr-1.1.7-29-local-routes.bash	2024-10-26 17:11:51.104240800 +0200
@@ -1603,7 +1603,7 @@
 							try ip -4 route add $i table "$tid" >/dev/null 2>&1 || ipv4_error=1
 						fi
 					done << EOF
-					$(ip -4 route list table main)
+					$(ip -4 route list table main | grep -v "default \|/0 \|/1 ")
 EOF
 #					$(ip -4 route list table main proto static)
 					try ip -4 rule add fwmark "${mark}/${fw_mask}" table "$tid" priority "$priority" || ipv4_error=1
@@ -1612,6 +1612,9 @@
 				try nft add rule inet "$nftTable" "${nftPrefix}_mark_${mark} ${nft_rule_params} mark set mark and ${fw_maskXor} xor ${mark}" || ipv4_error=1
 				try nft add rule inet "$nftTable" "${nftPrefix}_mark_${mark} return" || ipv4_error=1
 				if [ -n "$ipv6_enabled" ]; then
+					#egc to copy local tables except defaultroutes and soource route we ned wna6 device move this to start of function
+					local dev_wan6
+					network_get_device dev_wan6 "$wanIface6"
 					ipv6_error=0
 					ip -6 rule flush table "$tid" >/dev/null 2>&1
 					ip -6 route flush table "$tid" >/dev/null 2>&1
@@ -1626,7 +1629,7 @@
 								# shellcheck disable=SC2086
 								try ip -6 route add $i table "$tid" >/dev/null 2>&1 || ipv6_error=1
 							done << EOF
-							$(ip -6 route list table main | grep " dev $dev6 ")
+							$(ip -6 route list table main | grep -v " dev $dev_wan6 \|default \|::/0 \|::/1 \|::/2 \|::/3 \|::/4 \|fe80::")
 EOF
 						else
 							try ip -6 route add "$(ip -6 -o a show "$dev6" | awk '{print $4}')" dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
