--- pbr-1.1.7-9.bash	2024-09-11 14:49:49.370167800 +0200
+++ pbr-1.1.7-9-egc-1.bash	2024-09-11 14:52:04.460971495 +0200
@@ -1652,8 +1652,12 @@
 					ipv4_error=0
 					if [ -z "$gw4" ]; then
 						try ip -4 route add unreachable default table "$tid" >/dev/null 2>&1 || ipv4_error=1
-					else
+					elif is_wan "$iface"; then
+						echo -e "\negc: WAN interface=$iface\n"
 						try ip -4 route add default via "$gw4" dev "$dev" table "$tid" >/dev/null 2>&1 || ipv4_error=1
+					else
+						echo -e "\negc: OTHER interface=$iface\n"
+						try ip -4 route add default dev "$dev" table "$tid" >/dev/null 2>&1 || ipv4_error=1
 					fi
 # shellcheck disable=SC2086
 					while read -r i; do
@@ -1679,7 +1683,10 @@
 						if [ -z "$gw6" ] || [ "$gw6" = "::/0" ]; then
 							try ip -6 route add unreachable default table "$tid" >/dev/null 2>&1 || ipv6_error=1
 						elif ip -6 route list table main | grep -q " dev $dev6 "; then
-							ip -6 route add default via "$gw6" dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
+							#ip -6 route add default via "$gw6" dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
+							if ! ip -6 route add default via "$gw6" dev "$dev6" table "$tid" >/dev/null 2>&1; then
+								try ip -6 route add default dev "$dev6" table "$tid" metric 128 >/dev/null 2>&1 || ipv6_error=1  # might need metric to override default route
+							fi
 							while read -r i; do
 								i="$(echo "$i" | sed 's/ linkdown$//')"
 								i="$(echo "$i" | sed 's/ onlink$//')"
@@ -1691,6 +1698,7 @@
 						else
 							try ip -6 route add "$(ip -6 -o a show "$dev6" | awk '{print $4}')" dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
 							try ip -6 route add default dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
+							#try ip -6 route add default dev "$dev6" table "$tid" metric 128 >/dev/null 2>&1 || ipv6_error=1  # might need metric to override existing default route
 						fi
 					fi
 					try ip -6 rule add fwmark "${mark}/${fw_mask}" table "$tid" priority "$((priority-1))" >/dev/null 2>&1 || ipv6_error=1
@@ -1742,8 +1750,12 @@
 			if [ -n "$gw4" ] || [ "$strict_enforcement" -ne '0' ]; then
 				if [ -z "$gw4" ]; then
 					try ip -4 route add unreachable default table "$tid" >/dev/null 2>&1 || ipv4_error=1
-				else
+				elif is_wan "$iface"; then
+					echo -e "\negc: WAN interface=$iface\n"
 					try ip -4 route add default via "$gw4" dev "$dev" table "$tid" >/dev/null 2>&1 || ipv4_error=1
+				else
+					echo -e "\negc: OTHER interface=$iface\n"
+					try ip -4 route add default dev "$dev" table "$tid" >/dev/null 2>&1 || ipv4_error=1
 				fi
 				try ip rule add fwmark "${mark}/${fw_mask}" table "$tid" priority "$priority" || ipv4_error=1
 			fi
@@ -1762,6 +1774,7 @@
 					else
 						try ip -6 route add "$(ip -6 -o a show "$dev6" | awk '{print $4}')" dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
 						try ip -6 route add default dev "$dev6" table "$tid" >/dev/null 2>&1 || ipv6_error=1
+						#try ip -6 route add default dev "$dev6" table "$tid" metric 128 >/dev/null 2>&1 || ipv6_error=1  # might need metric to override existing default route
 					fi
 				fi
 				try ip -6 rule add fwmark "${mark}/${fw_mask}" table "$tid" priority "$priority" || ipv6_error=1
